From 215fa0c6d1e6de9e1a22105bbe728e1de614292d Mon Sep 17 00:00:00 2001
From: Jai Luthra <j-luthra@ti.com>
Date: Mon, 3 Jun 2024 13:06:16 +0530
Subject: [PATCH 325/465] media: cadence: csi2rx: Fix uninitialized used_vc
 variable

Initialize the used_vc variable to 0 to prevent lkp warning for
cases where source provides frame descriptor that does not populate
virtual channel for every stream.

Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202406011359.qVrQBfFU-lkp@intel.com/
Signed-off-by: Jai Luthra <j-luthra@ti.com>
---
 drivers/media/platform/cadence/cdns-csi2rx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/cadence/cdns-csi2rx.c b/drivers/media/platform/cadence/cdns-csi2rx.c
index 3138a7d3ac68..311afe9d2413 100644
--- a/drivers/media/platform/cadence/cdns-csi2rx.c
+++ b/drivers/media/platform/cadence/cdns-csi2rx.c
@@ -413,7 +413,7 @@ static void csi2rx_update_vc_select(struct csi2rx_priv *csi2rx,
 	/* If source provides per-stream VC info, use it to filter by VC */
 	for_each_active_route(&state->routing, route) {
 		int cdns_stream = route->source_pad - CSI2RX_PAD_SOURCE_STREAM0;
-		u8 used_vc;
+		u8 used_vc = 0;
 
 		for (i = 0; i < fd.num_entries; i++) {
 			if (fd.entry[i].stream == route->sink_stream) {
-- 
2.45.1

